clouddriver-saga/src/main/kotlin/com/netflix/spinnaker/clouddriver/saga/events.kt

```diff
@@ -60,6 +60,42 @@ class SagaLogAppended(
   )
 }
 
+class SagaSequenceUpdated(
+  sagaName: String,
+  sagaId: String,
+  val oldValue: Long,
+  val newValue: Long
+) : SagaEvent(sagaName, sagaId)
+
+class SagaRequiredEventsAdded(
+  sagaName: String,
+  sagaId: String,
+  val eventNames: List<String>,
+  val currentNumOfEvents: Int
+) : SagaEvent(sagaName, sagaId)
+
+class SagaRequiredEventsRemoved(
+  sagaName: String,
+  sagaId: String,
+  val eventNames: List<String>,
+  val currentNumOfEvents: Int
+) : SagaEvent(sagaName, sagaId)
+
+class SagaCompleted(
+  sagaName: String,
+  sagaId: String
+) : SagaEvent(sagaName, sagaId)
```

Comment:
We'll want a `val success: Boolean`. `SagaCompleted` will be fired at the completion of the compensation lifecycle as well, in which case the success value would be `false`.